src/main/java/de/tum/in/www1/artemis/web/rest/LearningPathResource.java

```diff
@@ -165,4 +165,37 @@ else if (!authorizationCheckService.isAtLeastInstructorInCourse(course, user) &&
         NgxLearningPathDTO graph = learningPathService.generateNgxGraphRepresentation(learningPath);
         return ResponseEntity.ok(graph);
     }
+
+    /**
+     * GET /courses/:courseId/learning-path-id : Gets the id of the learning path.
+     * If the learning path has not been generated although the course has learning paths enabled, the corresponding learning path will be created.
+     *
+     * @param courseId the id of the course from which the learning path id should be fetched
+     * @return the ResponseEntity with status 200 (OK) and with body the id of the learning path
+     */
+    @GetMapping("/courses/{courseId}/learning-path-id")
+    @EnforceAtLeastStudent
+    public ResponseEntity<Long> getLearningPathId(@PathVariable Long courseId) {
+        log.debug("REST request to get learning path id for course with id: {}", courseId);
+        Course course = courseRepository.findByIdElseThrow(courseId);
+        if (!authorizationCheckService.isStudentInCourse(course, null)) {
+            throw new BadRequestException("You are not a student in this course.");
+        }
+        if (!course.getLearningPathsEnabled()) {
+            throw new BadRequestException("Learning paths are not enabled for this course.");
```

Comment:
Is this really a client error? Maybe a 409 would be a better fit - it would indicate a request conflict with the current state of the course